### PR TITLE
fix(neon): update stride from px to bytes

### PR DESCRIPTION
**Pull request by PeterBee97**
### Description of the feature or fix
Fix NEON render error mentioned in #5514 

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html)
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

**Comment by PeterBee97** [comment]
We should really find a way to add asm acclerations to CI :(

**Review by FASTSHIFT — APPROVED** [review]

File: src/draw/sw/blend/neon/lv_blend_neon.S [hunk 1]
```diff
@@ -335,14 +335,14 @@ TMP_Q1      .qn  q14
...
-    sub             DST_STRIDE, DST_STRIDE, DST_W
     vmvn            D_A, OPA
 .if dst_bpp == 16
-    lsl             DST_STRIDE, DST_STRIDE, #1
+    sub             DST_STRIDE, DST_STRIDE, DST_W, lsl #1
 .elseif dst_bpp == 24
-    add             DST_STRIDE, DST_STRIDE, DST_STRIDE, lsl #1
+    sub             DST_STRIDE, DST_STRIDE, DST_W
+    sub             DST_STRIDE, DST_STRIDE, DST_W, lsl #1
 .elseif dst_bpp >= 31
```
Comment (XuNeo) [inline src/draw/sw/blend/neon/lv_blend_neon.S:344]: Is this a typo?

**Comment by kisvegabor** [comment]
> We should really find a way to add asm acclerations to CI :(

We are running all tests on Arm v7 too [might support Neon](https://developer.arm.com/documentation/dht0002/a/Introducing-NEON/What-is-NEON-). I don't know if our emulator really supports it or not.

cc @GorgonMeducer

**Comment by rmshub** [comment]
Tested in board with this fix. Unfortunately still issue present. Do I need to enable any configuration except below
```
#define  LV_USE_DRAW_SW_ASM     LV_DRAW_SW_ASM_NEON
```
Can we try SW blend with NEON intrinsics instead of pure Assembly language.

File: src/draw/sw/blend/neon/lv_blend_neon.S (hunk 1 above)
Comment (PeterBee97) [reply src/draw/sw/blend/neon/lv_blend_neon.S:344]: no :)

**Comment by PeterBee97** [comment]
> Tested in board with this fix. Unfortunately still issue present. Do I need to enable any configuration except below
> 
> ```
> #define  LV_USE_DRAW_SW_ASM     LV_DRAW_SW_ASM_NEON
> ```

Nothing else. I ran the tests on armv7m docker image and it works fine. Previous bug should cause segmentation faults most of the time since stride is larger than correct value so buffer will overflow. Are you using the latest lvgl and could you please provide more information about the case (code / what is expected to see)?

> Can we try SW blend with NEON intrinsics instead of pure Assembly language.

Intrinsics does not have macros, it would be tedious to write all blending functions one by one.

**Comment by rmshub** [comment]
I thought this PR is merged. My bad!. I will check it again after adding this fix patch

**Comment by rmshub** [comment]
@PeterBee97 After adding this fix patch. Neon rendering working as expected.

**Review by C47D — APPROVED** [review]
Approved based on feedback after applying this fix.

File: src/draw/sw/blend/neon/lv_blend_neon.S (hunk 1 above)
Comment (XuNeo) [reply src/draw/sw/blend/neon/lv_blend_neon.S:344]: `31` means for XRGB8888, and 32 for ARGB8888. So it's not a typo.
See 
https://github.com/lvgl/lvgl/blob/e4de2d265aa1b251038d18659835502e95d14eaf/src/draw/sw/blend/neon/lv_blend_neon.S#L681-L682

**Comment by kisvegabor** [comment]
> Neon rendering working as expected.

Can you compare the speed with and without neon using `lv_demo_benchmark`?

**Comment by GorgonMeducer** [comment]
> I don't know if our emulator really supports it or not.

Which emulator do you use?

**Comment by kisvegabor** [comment]
We are using [uraimo/run-on-arch-action@v2.6.0](https://github.com/lvgl/lvgl/blob/master/.github/workflows/ccpp.yml#L75C15-L75C47)

**Pull request merged**